back-end/API/app/controllers/enterprise_controller.py

```
@@ -38,33 +38,34 @@ def getEnterpriseByEmployer(id):
+        if current_user.isModerator and data.get("id") is not None:
             enterprise = enterprise_service.getEnterprise(data["id"])
             if enterprise:
                 enterprise_service.updateEnterprise(data)
                 return jsonify({'message': 'enterprise updated'})
+            else:
+                logger.warn(f'Enterprise not found with id : {data["id"]}')
+                return jsonify({'message': 'enterprise not found'}), 404
         else:
+            employer = employer_service.getEmployerByUserId(current_user.id)
+            if not employer:
+                logger.warn("User has no associated employer")
```

Ruff (0.15.2)

[warning] Logging statement uses `warn` instead of `warning`

Convert to `warning`

(G010)